oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/ModelService.kt

```diff
@@ -0,0 +1,68 @@
+package com.empowerops.volition.ref_oasis
+
+import com.empowerops.volition.dto.ErrorResponseDTO
+import com.empowerops.volition.dto.NodeStatusCommandOrResponseDTO
+import com.empowerops.volition.dto.OASISQueryDTO
+import com.empowerops.volition.dto.SimulationResponseDTO
+import io.grpc.stub.StreamObserver
+import javafx.beans.property.SimpleStringProperty
+import javafx.collections.ObservableList
+import kotlinx.coroutines.channels.Channel
+import java.time.Duration
+import java.time.LocalDateTime
+
+interface ModelService {
+    var simByName: Map<String, Simulation>
+    fun updateStatusMessage(message: String)
+    fun addMessage(message: Message)
+    fun addResult(result: Result)
+    fun setDuration(nodeName: String?, timeOut: Duration?)
+
+    fun updateSim(name: String, newNode: Simulation)
+    fun addNewSim(name: String, simulation: Simulation)
+    fun removeSim(name: String)
+    fun renameSim(target: Simulation, newName: String, oldName: String)
+}
+
```

Comment:
Nice, so this is effectively your event bus, and it doesn't depend on java-fx. Can you move this interface into its own file so that it doesn't. `import javafx...`?

Also, I would encourage you to give these things event sounding names:

```kt
interface ModelService {
  fun onStatusChanged(statusMessage: String)
  fun onMessageEncounter(message: Message)
  fun onResultCreated(result: Result)
}
```

The idea is that we want to express these functions as being notifications of something happening, not a request to do something. Ideally our code should be structured such that it does not care who is listening --much like our optimizer in OASIS. 

Finally, I would also encourage you to give it a companion object that doesnt do anything:

```kt
interface ModelService {
  fun updateStatusMessage(message: String)
  fun addMessage(message: Message)
  //...
}
```

Reply:
I didn't address these but I will try to take another look in next issue, but in general, the method name need another look. I do move the file.